db/database.go

```diff
@@ -569,7 +569,14 @@ func (context *DatabaseContext) RemoveObsoleteDesignDocs(previewOnly bool) (remo
 
 // Removes previous versions of Sync Gateway's design docs found on the server
 func (context *DatabaseContext) RemoveObsoleteIndexes(previewOnly bool) (removedIndexes []string, err error) {
-	return removeObsoleteIndexes(context.Bucket, previewOnly, context.UseXattrs())
+
+	gocbBucket, ok := base.AsGoCBBucket(context.Bucket)
+	if !ok {
+		base.Warnf(base.KeyAll, "Cannot remove obsolete indexes for non-gocb bucket - skipping.")
+		return removedIndexes, nil
```

Comment:
Do we need to create the zero-length slice to prevent it being null like `removeObsoleteIndexes` did?